File: 746.Min-Cost-Climbing-Stairs/dp.py
class Solution:
    def minCostClimbingStairs(self, cost: list[int]) -> int:
        a, b = 0, cost[0]

        # O(n)
        for i in range(1, len(cost)):
            a, b = b, cost[i] + min(a, b)

        return min(a, b)

File: 746.Min-Cost-Climbing-Stairs/dp.rs
impl Solution {
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let (mut a, mut b) = (0, cost[0]);

        // O(n)
        for &v in cost.iter().skip(1) {
            let curr_cost = v + a.min(b);
            (a, b) = (b, curr_cost);
        }

        a.min(b)
    }
}
